core/quantity/src/test/Parsing.test.ts

```diff
@@ -138,6 +138,52 @@ describe("Parsing tests:", () => {
     }
   });
 
+  it("Generate Parse Tokens from composite string with dash spacer without math operations allowed", async () => {
+    const formatData = {
+      composite: {
+        includeZero: true,
+        spacer: "-",
+        units: [
+          {
+            label: "'",
+            name: "Units.FT",
+          },
+          {
+            label: `"`,
+            name: "Units.IN",
+          },
+        ],
+      },
+      decimalSeparator: ".",
+      formatTraits: [
+        "KeepSingleZero",
+        "ShowUnitLabel",
+      ],
+      precision: 8,
+      roundFactor: 0,
+      showSignOption: "OnlyNegative",
+      type: "Fractional",
+      uomSeparator: "",
+      allowMathematicOperations: false,
+    };
+    const format = new Format("test");
+    const unitsProvider = new TestUnitsProvider();
+    await format.fromJSON(unitsProvider, formatData).catch(() => { });
+
+    const tests = [
+      {input: "12'-6 1/2\"", expectedTokens: [{ value: 12 }, { value: "'" }, { value: 6.5 }, { value: '"' }]},
+      {input: "-2FT-6IN + 6IN", expectedTokens: [{value: "-", isOperand: true}, { value: 2 }, { value: "FT" }, { value: 6 }, { value: "IN" }, { value: "+", isOperand: true }, { value: 6 }, { value: "IN" }]},
```

Comment:
```suggestion
      { input: "12'-6 1/2\"", expectedTokens: [{ value: 12 }, { value: "'" }, { value: 6.5 }, { value: '"' }]},
      { input: "-2FT-6IN + 6IN", expectedTokens: [{ value: "-", isOperand: true }, { value: 2 }, { value: "FT" }, { value: 6 }, { value: "IN" }, { value: "+", isOperand: true }, { value: 6 }, { value: "IN" }]},
```

Reply:
Please, don't nitpick whitespace. File an issue to address lack of automated code formatting.